grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1411,4 +1411,15 @@ class ProductService {
         }
         return product
     }
+
+    Synonym editProductSynonym(String synonymId, String synonymTypeCodeName, String synonymValue, String localeName) {
+        Synonym synonym = Synonym.get(synonymId)
+        Locale locale = localeName ? new Locale(localeName) : null
+        SynonymTypeCode synonymTypeCode = synonymTypeCodeName ? SynonymTypeCode.valueOf(synonymTypeCodeName) : SynonymTypeCode.ALTERNATE_NAME
```

Comment:
Can you provide some rationale for this? Something like "it's in the requirements, dummy" would suffice. 

If this was just an assumption you made, I might prefer throwing an exception if the synonym type code is not specified. But would like to hear from others as well.

Reply:
No ti was not in the requirements, this was just my assumption based on the **addSynonymToProduct** which defaults to ALTERNATE_NAME.
I agree, in case of an edit it should probably throw an exception